router/health_check.go

```diff
@@ -0,0 +1,123 @@
+package router
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+
+	"github.com/buildwithgrove/path/relayer/morse"
+	"github.com/buildwithgrove/path/relayer/shannon"
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+// HealthCheck is an interface that must be implemented
+// by components that need to report their health status
+type HealthCheck interface {
+	// Name returns the name of the component being checked.
+	Name() string
+	// IsAlive returns true if the component is healthy, otherwise false.
+	IsAlive() bool
+}
+
+// All components that implement the HealthCheck interface must be registered here.
+var _ HealthCheck = &morse.Protocol{}
+var _ HealthCheck = &shannon.Protocol{}
+
+// One of `ready` or `not_ready`.
+type healthCheckStatus string
+
+const (
+	// StatusReady indicates that all PATH components are ready
```

Comment:
```suggestion
	// StatusReady indicates that the PATH component is ready
```

Reply:
I think indicating that it refers to multiple components being ready to serve relays is more accurate. Currently it is only the Protocol but in the future could be extended to many components that need to signal they are ready, eg. QoS.